### PR TITLE
Add behaviour tables for `GetResponse` and `SubscribeResponse`

### LINKED ISSUE #142: What should a gNMI server return when a default value is queried?
There are four independent cases considered below. All examples use the /interfaces/interface/state/enabled leaf, which has a default value of `true`.

## Question
For all queries below, consider the case that the interface at /interfaces/interface[name="foo"] exists, and that its "enabled" field has its default value of `true`.

The question for all of them is, must ("/interfaces/interface[name="foo"]/state/enabled", `true`) be returned by GET?

### 1 Non-wildcard leaf query
e.g. GET /interfaces/interface[name="foo"]/state/enabled

### 2 Non-wildcard non-leaf query
e.g. GET /interfaces/interface[name="foo"]

### 3 Wildcard leaf query
e.g. GET /interfaces/interface[name="*"]/state/enabled

### 4 Wildcard non-leaf query
e.g. GET /interfaces/interface[name="*"]

### PR DESCRIPTION
**Pull request by wenovus**
Fixes #142 and #155

These were extracted from the comments within the issues listed above, especially https://github.com/openconfig/reference/issues/155#issuecomment-1154173549

### CODE REVIEW

**Comment by wenovus** [comment]
One thing that's not clear yet about this PR is https://github.com/openconfig/reference/issues/155#issuecomment-1176835181

File: rpc/gnmi/gnmi-specification.md [hunk 1]
```diff
@@ -898,6 +900,15 @@ accuracy for individual data items, the `Subscribe` RPC is recommended to
 request a telemetry stream (see [Section
 3.5.2](#352-sending-telemetry-updates)).
 
+### 3.3.4 GetResponse Behavior Table
+
+| GetRequest Scenario                                                                                                      | Target Behavior |
+| ------------------------------------------------------------------------------------------------------------------------ | ---------------- |
+| Subscribed paths exist or a YANG default value is [in use](https://datatracker.ietf.org/doc/html/rfc7950#section-7.6.1). | Value(s) are returned |
+| Subscribed paths are syntactically correct but one or more paths do not (yet) exist.                                     | Return `NOT_FOUND` |
```
Comment (hellt) [inline rpc/gnmi/gnmi-specification.md:908]: should it further clarify here as `... do not (yet) exist or default value is not in use`?

File: rpc/gnmi/gnmi-specification.md [hunk 2]
```diff
@@ -898,6 +900,15 @@ accuracy for individual data items, the `Subscribe` RPC is recommended to
 request a telemetry stream (see [Section
 3.5.2](#352-sending-telemetry-updates)).
 
+### 3.3.4 GetResponse Behavior Table
+
```
[the remark this answers is not in the document]
Comment (wenovus) [reply rpc/gnmi/gnmi-specification.md:904]: Done. PTAL

File: rpc/gnmi/gnmi-specification.md [hunk 3]
```diff
@@ -898,6 +900,15 @@ accuracy for individual data items, the `Subscribe` RPC is recommended to
...
+| GetRequest Scenario                                                                                                      | Target Behavior |
+| ------------------------------------------------------------------------------------------------------------------------ | ---------------- |
+| Subscribed paths exist or a YANG default value is [in use](https://datatracker.ietf.org/doc/html/rfc7950#section-7.6.1). | Value(s) are returned |
```
[the remark this answers is not in the document]
Comment (wenovus) [reply rpc/gnmi/gnmi-specification.md]: Changed to "requested paths"

File: rpc/gnmi/gnmi-specification.md [hunk 4]
```diff
@@ -53,14 +54,15 @@ April 28, 2022
       * [3.5.1.1 The SubscribeRequest Message](#3511-the-subscriberequest-message)
       * [3.5.1.2 The SubscriptionList Message](#3512-the-subscriptionlist-message)
       * [3.5.1.3 The Subscription Message](#3513-the-subscription-message)
-      * [3.5.1.4 The SubscribeResponse Message](#3514-the-subscriberesponse-message)
+  * [3.5.1.4 The SubscribeResponse Message](#3514-the-subscriberesponse-message)
```
[the remark this answers is not in the document]
Comment (wenovus) [reply rpc/gnmi/gnmi-specification.md]: Thanks, fixed.

File: rpc/gnmi/gnmi-specification.md (hunk 1 above)
Comment (wenovus) [reply rpc/gnmi/gnmi-specification.md]: Thanks, expanded sentence in both tables.

**Review by robshakir — APPROVED** [review]

**Comment by wenovus** [comment]
@robshakir Could you also take a look at https://github.com/openconfig/reference/issues/155#issuecomment-1176835181 and the discussion following it I had with Darren? Depending on how that's resolved this table could change.

**Pull request merged**